Research on stake delegation

eth-r opened the pull request:
Investigate options for stake delegation, their security implications, and the
complexity of implementation.

eth-r committed:
Add research on stake delegation

Investigate options for stake delegation, their security implications, and the
complexity of implementation.

mhluongo reviewed (dismissed):
Awesome overview of our options

eth-r committed:
Rename magpie to beneficiary

The role of the address receiving rewards from stakers' work was previously
named as "magpie" but the name is non-obvious and would be better used for
stakers who seek easy rewards by e.g. trying to discover misbehavior and submit
proofs of it to receive tattletale rewards.

eth-r committed:
Add partial redelegation and thorough explanation of load-balancing

Whether stakes can be partially redelegated has a significant impact on
scalability, but partial redelegation is effectively a more convenient way of
blitzpantsing and operation-specific sub-operators have a significant advantage
over redelegation-based load balancing, so the conclusion remains unchanged.

mhluongo reviewed (approved):
Looks good. I don't like the complex implications on system design but research isn't to report things we like :smile:
